spec/index.bs

```diff
@@ -11226,6 +11257,38 @@ It must only be included by interfaces which also include those mixins.
     Otherwise return `true`.
 </div>
 
+<div algorithm data-timeline=device>
+    <dfn abstract-op>Validate immediate data</dfn>(encoder, pipeline)
+
+    **Arguments:**
+
+    : {{GPUBindingCommandsMixin}} |encoder|
+    :: Encoder whose immediate data is being validated.
+    : {{GPUPipelineBase}} |pipeline|
+    :: Pipeline to validate |encoder|'s immediate data is compatible with.
+
+    [=Device timeline=] steps:
+
+    1. If |immediateSize| is 0, return `true`.
+    2. If any of the following conditions are unsatisfied, return `false`:
+
+        <div class=validusage>
+            - |pipeline| must not be `null`.
+            - Let |pipelineLayout| be |pipeline|.{{GPUPipelineBase/[[layout]]}}.
+            - Let |immediateSize| be |pipelineLayout|.{{GPUPipelineLayout/[[immediateSize]]}}.
+            - For each immediate data variable that is [=statically used=] by any entry point in |pipeline|:
```

Comment:
When I read this I think "immediate data variable" means one `var<immediate>`, of which there is of course only one (at most). But that `var` itself has a `SizeOf` the entire struct, whereas we actually want only *accessible* parts of the struct to need to be initialized. That is going to require the WGSL spec to "reflect" some new information for us - the map of `slot -> is_accessible` - for us to validate against.

I would recommend combining this PR with the WGSL PR, since there will be new integration between the specs.